### DISCUSSION TITLE
请问如果自己想升级使用中的V8版本，有什么注意事项？

### DISCUSSION THREAD

**Discussion by zhaojunmeng** [Tencent/puerts#1396] [Q&A]
比如我想使用比V8 9.4版本更新的V8版本，请问自己升级的话，有什么需要注意的事项吗？

或者说PuerTS有尝试过升级到更新的版本，有没有遇到什么坑之类的。

**Comment by zombieyang** [comment]
https://github.com/zombieyang/puerts/commit/93c8c091a29657d11204407409774645ff8be732
可以参考下这个，之前稍微搞过。目前还没有提上合入主干的日程

**Reply by zhaojunmeng** [reply]
请问有Unreal版本相关升级的提交可供参考吗？（标题漏写明是特指哪个引擎了）

**Reply by chexiongsheng** [reply]
最新release有9.4的支持，对puerts其实就替换包而已。
关键在于v8的编译：按我们的经验，并不是所有版本都支持各平台。其次，你得把我们对v8的patch同步到你要的v8版本。最后，也不是所有v8版本都能和ue引擎一起工作的。

**Reply by chexiongsheng** [reply]
关键在于和ue的兼容性，即使9.4，也是有项目用了比较久的一段时间，确认和ue没冲突了，我才在v1.0.4附上。

**Reply by zhaojunmeng** [reply]
v8的各个平台编译，我已经跑了一遍了，目前有经验了
目前看起来，主要是：“和ue兼容性”这里，是需要花时间测试的。

那请问有之前一些已知的，和ue不兼容产生的副作用表现吗？

**Comment by chexiongsheng** [comment]
> v8的各个平台编译，我已经跑了一遍了，目前有经验了 目前看起来，主要是：“和ue兼容性”这里，是需要花时间测试的。
> 
> 那请问有之前一些已知的，和ue不兼容产生的副作用表现吗？

比如ndk版本不兼容，c++库冲突之类。

**Reply by zhaojunmeng** [reply]
了解了，感谢
